player.gd

```diff
@@ -72,10 +73,13 @@ var time_sliding: float = 0.0
 
 var dead: bool = false
 
-func respawn(spawn_point: SpawnPoint = respawn_point) -> void:
+func respawn(point: RespawnPoint = respawn_point) -> void:
 	death_particles.hide()
 	death_particles.emitting = false
-	spawn_point.teleport(self)
+	if point != null:
+		point.spawn_point.teleport(self)
+	else:
+		spawn_point.teleport(self)
 	animation_manager.show()
 	death_text.hide()
 	dead = false
```

Comment:
add at end of code, make sure to set velocity to 0 after respawning!
`velocity.x = 0.0`
`velocity.y = 0.0`

Reply:
teehee